api/src/test/java/org/apache/iceberg/geospatial/TestGeospatialPredicateEvaluators.java

```diff
@@ -0,0 +1,542 @@
+    BoundingBox box1 = new BoundingBox(min1, max1);
+
+    GeospatialBound min2 = GeospatialBound.createXY(2.0, 2.0);
+    GeospatialBound max2 = GeospatialBound.createXY(5.0, 5.0);
+    BoundingBox box2 = new BoundingBox(min2, max2);
+
+    assertThat(evaluator.intersects(box1, box2)).isTrue();
+    assertThat(evaluator.intersects(box2, box1)).isTrue();
+  }
+
+  @Test
+  public void testBoxesWithZCoordinate() {
+    Type geometryType = Types.GeometryType.crs84();
+    GeospatialPredicateEvaluators.GeospatialPredicateEvaluator evaluator =
+        GeospatialPredicateEvaluators.create(geometryType);
+
+    // Two boxes with Z coordinates that overlap in X and Y but not in Z
+    GeospatialBound min1 = GeospatialBound.createXYZ(0.0, 0.0, 0.0);
+    GeospatialBound max1 = GeospatialBound.createXYZ(2.0, 2.0, 1.0);
+    BoundingBox box1 = new BoundingBox(min1, max1);
+
+    GeospatialBound min2 = GeospatialBound.createXYZ(1.0, 1.0, 2.0);
+    GeospatialBound max2 = GeospatialBound.createXYZ(3.0, 3.0, 3.0);
+    BoundingBox box2 = new BoundingBox(min2, max2);
+
+    GeospatialBound min3 = GeospatialBound.createXYZ(1.0, 1.0, 1.0);
+    GeospatialBound max3 = GeospatialBound.createXYZ(3.0, 3.0, 3.0);
+    BoundingBox box3 = new BoundingBox(min3, max3);
+
+    assertThat(evaluator.intersects(box1, box2)).isFalse();
+    assertThat(evaluator.intersects(box2, box1)).isFalse();
+    assertThat(evaluator.intersects(box1, box3)).isTrue();
+    assertThat(evaluator.intersects(box3, box1)).isTrue();
+    assertThat(evaluator.intersects(box2, box3)).isTrue();
+    assertThat(evaluator.intersects(box3, box2)).isTrue();
+  }
+
+  @Test
+  public void testBoxesWithMCoordinate() {
+    Type geometryType = Types.GeometryType.crs84();
+    GeospatialPredicateEvaluators.GeospatialPredicateEvaluator evaluator =
+        GeospatialPredicateEvaluators.create(geometryType);
+
+    // Two boxes with M coordinates that overlap in X and Y but not in M
+    GeospatialBound min1 = GeospatialBound.createXYM(0.0, 0.0, 0.0);
+    GeospatialBound max1 = GeospatialBound.createXYM(2.0, 2.0, 1.0);
+    BoundingBox box1 = new BoundingBox(min1, max1);
+
+    GeospatialBound min2 = GeospatialBound.createXYM(1.0, 1.0, 2.0);
+    GeospatialBound max2 = GeospatialBound.createXYM(3.0, 3.0, 3.0);
+    BoundingBox box2 = new BoundingBox(min2, max2);
+
+    GeospatialBound min3 = GeospatialBound.createXYM(1.0, 1.0, 1.0);
+    GeospatialBound max3 = GeospatialBound.createXYM(3.0, 3.0, 3.0);
+    BoundingBox box3 = new BoundingBox(min3, max3);
+
+    assertThat(evaluator.intersects(box1, box2)).isFalse();
+    assertThat(evaluator.intersects(box2, box1)).isFalse();
+    assertThat(evaluator.intersects(box1, box3)).isTrue();
+    assertThat(evaluator.intersects(box3, box1)).isTrue();
+    assertThat(evaluator.intersects(box2, box3)).isTrue();
+    assertThat(evaluator.intersects(box3, box2)).isTrue();
+  }
+
+  @Test
+  public void testGeometryWrapAroundOnA() {
```

Comment:
all test methods before this are for geometry. should we add geometry to the method names, like `testGeometryBoxesWithMCoordinate` for line 158 above?